vaporetto/src/sentence.rs

```diff
@@ -278,964 +318,900 @@ impl Sentence {
-        &self.chars
+    #[inline]
+    pub fn tags(&self) -> &[Option<Cow<'b, str>>] {
+        &self.tags
     }
 
-    /// Gets immutable references to the characters and character types, and a mutable reference to
-    /// boundaries.
-    ///
-    /// # Returns
-    ///
-    /// A tuple of references.
+    /// Returns a mutable reference to the internal representation of tags.
     ///
     /// # Examples
     ///
     /// ```
-    /// use vaporetto::{BoundaryType, CharacterType, Sentence};
+    /// use vaporetto::Sentence;
     ///
-    /// let mut s = Sentence::from_partial_annotation("A-1|あ エ-漢|?").unwrap();
-    /// let (chars, char_types, boundaries) = s.chars_and_boundaries_mut();
-    /// assert_eq!(&['A', '1', 'あ', 'エ', '漢', '?'], chars);
-    /// assert_eq!(&[
-    ///     CharacterType::Roman as u8,
-    ///     CharacterType::Digit as u8,
-    ///     CharacterType::Hiragana as u8,
-    ///     CharacterType::Katakana as u8,
-    ///     CharacterType::Kanji as u8,
-    ///     CharacterType::Other as u8,
-    /// ], char_types);
-    /// assert_eq!(&[
-    ///     BoundaryType::NotWordBoundary,
-    ///     BoundaryType::WordBoundary,
-    ///     BoundaryType::Unknown,
-    ///     BoundaryType::NotWordBoundary,
-    ///     BoundaryType::WordBoundary,
-    /// ], boundaries);
+    /// let mut s = Sentence::from_tokenized("火星/名詞/カセー に 行き/動詞 まし/助動詞/マシ た").unwrap();
+    /// s.tags_mut()[4].replace("助詞".into());
+    /// let mut buf = String::new();
+    /// s.write_tokenized_text(&mut buf);
+    /// assert_eq!("火星/名詞/カセー に/助詞 行き/動詞 まし/助動詞/マシ た", buf);
     /// ```
-    pub fn chars_and_boundaries_mut(&mut self) -> (&[char], &[u8], &mut [BoundaryType]) {
-        (&self.chars, &self.char_type, &mut self.boundaries)
+    #[inline]
+    pub fn tags_mut(&mut self) -> &mut [Option<Cow<'b, str>>] {
+        &mut self.tags
     }
 
-    /// Gets a reference to the character type information.
-    ///
-    /// # Returns
-    ///
-    /// A reference to the character type information.
+    /// Update the tag information.
+    /// If you want to predict tags, call this function after calling [`Predictor::predict()`] and
+    /// word boundaries are fixed.
+    ///
+    #[cfg_attr(
+        feature = "std",
+        doc = "
+# Examples
+
+```
+use std::fs::File;
+
+use vaporetto::{Model, Predictor, Sentence};
+
+let f = File::open(\"../resources/model.bin\").unwrap();
+let model = Model::read(f).unwrap();
+let predictor = Predictor::new(model, true).unwrap();
+
+let mut s = Sentence::from_raw(\"まぁ良いだろう\").unwrap();
+predictor.predict(&mut s);
+let mut buf = String::new();
+s.write_tokenized_text(&mut buf);
+assert_eq!(\"まぁ 良い だろう\", buf);
+
+s.fill_tags();
+
+s.write_tokenized_text(&mut buf);
+assert_eq!(
+    \"まぁ/副詞/マー 良い/形容詞/ヨイ だろう/助動詞/ダロー\",
+    buf,
+);
+```
+"
+    )]
+    #[cfg(feature = "tag-prediction")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "tag-prediction")))]
+    #[inline]
+    pub fn fill_tags(&mut self) {
```

Comment:
Why do we need this function?

This involves the cross-reference between `sentence.rs` and `predictor.rs`, resulting in complicating the modules.
I think `Predictor::predict_tags` is sufficient, and it is preferable not to have this function.

Reply:
This implementation ensures that the Predictor used for filling tags is identical with the one used for score calculation.

If `Predictor::predict_tags()` is exposed, users can call that function with inappropriate Predictor and Sentence pairs.

Reply:
Okay